.buildkite/scripts/commit.sh

```diff
@@ -16,35 +22,36 @@ if [ -z "${BUILDKITE}" ]; then
 	exit 1
 fi
 
-if [ "${BUILDKITE_BRANCH}" = "main" ]; then
-	echo "Skipping commit to main"
+# Commits only need to be done if:
+# (1) this is a PR build
+# (2) the quark submodule has changed compared to the base branch
+if [ -z "${BUILDKITE_PULL_REQUEST}" ]; then
+	echo "Skipping commit for non-PR build"
 	exit 0
-fi 
+fi
+
+if [ "$(git diff --submodule ${BUILDKITE_PULL_REQUEST_BASE_BRANCH} | wc -c)" = "0" ]; then
+	echo "Skipping commit; no change detected in quark submodule"
```

Comment:
I think we can use `git diff --submodule --exit-code --quiet` here
```
eru:go-quark: git diff --submodule --exit-code; echo $?
0

eru:go-quark: echo foo >src/kprobe_queue.c
eru:go-quark: git diff --submodule --exit-code; echo $?
Submodule src contains modified content
1
```
```suggestion
if git diff --exit-status --quiet --submodule ${BUILDKITE_PULL_REQUEST_BASE_BRANCH}; then
	echo "Skipping commit; no change detected in quark submodule"
```